docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,213 @@
+
+Although many details were glossed over (such as how "borrowing" compute time on a thread pool is coordinated), the important thing to recognize here is that **no thread is 100% dedicated to running the initiated task**.  This allows threads in the thread pool of a system to handle a larger volume of work rather than having to wait for I/O to finish.
+
+Tasks are awaitable, meaning that the use ``await`` will allow your application or service to perform useful work while the task is running by yielding control to its caller until the task is done.  If you're using ``Task<T>``, the ``await`` keyword will additionally "unwrap" the value returned when the Task is complete.
+
+Although the above may seem like a lot of work to be done, when measured in terms of execution time, it's miniscule compared to the time it takes to do the actual I/O work.  Although not at all precise, a potential timeline for such a call would look like this:
+
+0-1--------------------------------------------------------------------------------------------------------------------------------------------------2-3
+
+* Time spent from points ``0`` to ``1`` is everything up until an async method yield control to its caller.
+
+* Time spent from points ``1`` to ``2`` is the time spend on I/O.
+
+* Finally, time spent from ``2`` to ``3`` is passing control back to the async method, at which point it is executing again.
+
+Tasks are also used outside of the async programming model.  They are the foundation of the Task Parallel Library, which supports the parallelization of CPU-bound work via `Data Parallelism <https://msdn.microsoft.com/en-us/library/dd537608(v=vs.110).aspx>`_ and `Task Parallelism <https://msdn.microsoft.com/en-us/library/dd537609(v=vs.110).aspx>`_.
+
+Important Info and Advice
+-------------------------
+
+Although async programming is relatively straightforward, there are some details to keep in mind which can prevent unexpected behavior.
+
+* **You should add "Async" as the suffix of every async method name you write.**
+
+This is the convention used in .NET to more-easily differentiate synchronous and asynchronous methods.  Note that certain methods which aren't explicitly called by your code (such as event handlers or web controller methods) don't necessarily apply.  Because these are not explicitly called by your code, being explicit about their naming isn't as important.
+
+* ``async void`` **should only be used for event handlers.**
+
+Why?  It's the only way to allow asynchronous event handlers work because events do not have return types (thus cannot make use of ``Task`` and ``Task<T>``).  Any other use of ``async void``does not follow the Task-based model and will bring about some issues, such as:
```

Comment:
"Why?" isn't really needed. You don't use that for the first statement.

"handlers to work"